docs/overloading-wp-behavior.md

```diff
@@ -4,6 +4,139 @@ SnapWP takes an additive approach to headless WordPress development, where WordP
 
 This tutorial explains how to overload default WordPress behavior using SnapWP, focusing on customizing blocks, creating custom frontend routes, and `html-react-parser` component mapping.
 
+## Default Block Rendering
+
+This tutorial covers how SnapWP uses the `Parse` component to render HTML content. The `Parse` component converts HTML strings into React components, which is essential for displaying default WordPress blocks.
+
+### Using `Parse`
+
+The `Parse` component takes an HTML string (`renderedHtml`) and converts it into React components. Here's a basic example:
+
+```tsx
+import React from 'react';
+import { Parse } from '@snapwp/next';
+import { type BlockData } from '@snapwp/core';
+
+/**
+ * Renders the default block.
+ * @param props - The props for the component.
+ * @param props.renderedHtml - The rendered HTML.
+ * @return The rendered default block.
+ */
+export default function Default( { renderedHtml }: BlockData ) {
+	return <Parse html={ renderedHtml || '' } />;
+}
+```
+
+## Overloading [html-react-parser](https://www.npmjs.com/package/html-react-parser) Options
+
+SnapWP allows you to pass custom options to [**react-parser**](../packages/next/src/react-parser/index.tsx) Component which internally uses html-react-parser. This is useful when you want to overload default options of html-react-parser.
+
+### 1. Creating a Custom Component with options
+
+Create a new Custom Component to overload the default options of [react-parser](../packages/next/src/react-parser/index.tsx).
+
+MyCustomReactParser.tsx
+
+```tsx
+import React from 'react';
+import {
+	Element,
+	type DOMNode,
+	type HTMLReactParserOptions,
+} from 'html-react-parser';
+import { CustomImage } from '@/CustomImage'; // custom image component defined in your project
+/**
+ *  Using typescript type guard to check if a DOMNode is an Element.
+ * Ref : https://github.com/remarkablemark/html-react-parser/issues/221#issuecomment-784073240
+ * @param domNode The DOM element
+ * @return parser options
+ */
+const isElement = ( domNode: DOMNode ): domNode is Element => {
+	const isTag = domNode.type === 'tag';
+	const hasAttributes = ( domNode as Element ).attribs !== undefined;
+
+	return isTag && hasAttributes;
+};
```

Comment:
Should we be doing this in _our_ `options`? If so, makes sense to export this function and update the docs (in a follow up PR)